query-engine/core/src/response_ir/internal.rs

```diff
@@ -296,7 +296,7 @@ fn finalize_objects(
                         Ok((parent, Item::Ref(ItemRef::new(first))))
                     } else if items.is_empty() && opt {
                         Ok((parent, Item::Ref(ItemRef::new(Item::Value(PrismaValue::Null)))))
-                    } else if items.is_empty() && opt {
```

Comment:
Notice how these two conditions were identical, and hence the second one would never be conditionally chosen.